internal/location/coordinates.go

```diff
@@ -0,0 +1,112 @@
+package location
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"math/rand"
+	"net/http"
+	"os"
+
+	mapbox "github.com/ryankurte/go-mapbox/lib"
+	"github.com/ryankurte/go-mapbox/lib/geocode"
+)
+
+const geocodeRequestLimit = 1
+const zipCodeTokenKey = "ZIP_CODE_TOKEN"
+const zipCodeURL = "https://app.zipcodebase.com/api/v1/code/city?apikey=%s&city=%s&country=ca&limit=%d"
+
+type zipcodeResponse struct {
+	Results []string `json:"results"`
+}
+
+type coordinatesFinder struct {
+	mapbox       *mapbox.Mapbox
+	forwardOpts  geocode.ForwardRequestOpts
+	zipCodeToken string
+}
+
+func NewCoordinatesFinder(token string, country string) (LocationFinder, error) {
+	mapbox, err := mapbox.NewMapbox(token)
+	if err != nil {
+		return nil, err
+	}
+
+	forwardOpts := geocode.ForwardRequestOpts{
+		Limit:   geocodeRequestLimit,
+		Country: country,
+	}
+
+	zipCodeToken := os.Getenv(zipCodeTokenKey)
+
+	return coordinatesFinder{
+		mapbox:       mapbox,
+		forwardOpts:  forwardOpts,
+		zipCodeToken: zipCodeToken,
+	}, err
+}
+
+func (finder coordinatesFinder) GetCityCenter(city string) (Coordinates, error) {
+	forward, err := finder.mapbox.Geocode.Forward(city, &finder.forwardOpts)
+	if err != nil {
+		return Coordinates{}, err
+	}
+	if len(forward.Features) == 0 {
+		return Coordinates{}, errors.New("Not a valid city")
+	}
+
+	return Coordinates{
+		Latitude:  forward.Features[0].Center[1],
+		Longitude: forward.Features[0].Center[0],
+	}, nil
+}
+
+func randomCoords(coordinate float64) float64 {
+	num := math.Floor(rand.Float64()*99) + 1
+	if math.Round(rand.Float64()) == 1 {
+		num *= 1
```

Comment:
we multiply the number... by 1? lol

Reply:
lol idk i didnt write this just moved it from the other file